src/pages/index.tsx

```diff
@@ -1,28 +1,94 @@
 import { EventCarousel } from '@/components/events';
-import { EventAPI } from '@/lib/api';
+import Hero from '@/components/home/Hero';
+import { showToast } from '@/lib';
+import { EventAPI, UserAPI } from '@/lib/api';
 import withAccessType from '@/lib/hoc/withAccessType';
+import { attendEvent } from '@/lib/managers/EventManager';
 import { CookieService, PermissionService } from '@/lib/services';
-import type { PublicAttendance, PublicEvent } from '@/lib/types/apiResponses';
+import type {
+  CustomErrorBody,
+  PrivateProfile,
+  PublicAttendance,
+  PublicEvent,
+} from '@/lib/types/apiResponses';
 import { CookieType } from '@/lib/types/enums';
 import styles from '@/styles/pages/Home.module.scss';
 import { GetServerSideProps } from 'next';
+import { useEffect, useState } from 'react';
 
 interface HomePageProps {
+  user: PrivateProfile;
+  authToken: string;
   pastEvents: PublicEvent[];
   upcomingEvents: PublicEvent[];
   liveEvents: PublicEvent[];
   attendances: PublicAttendance[];
+  checkInResponse: PublicEvent | CustomErrorBody | null;
 }
 
-const PortalHomePage = ({ pastEvents, upcomingEvents, liveEvents, attendances }: HomePageProps) => {
+const processCheckInResponse = (
+  response: PublicEvent | CustomErrorBody
+): PublicEvent | undefined => {
+  if ('uuid' in response) {
+    // If the response contains a uuid, the response is a PublicEvent.
+    const title = `Checked in to ${response.title}!`;
+    const subtitle = `Thanks for checking in! You earned ${response.pointValue} points.`;
+    showToast(title, subtitle);
+    return response;
+  }
+  showToast('Unable to checkin!', response.message);
+  return undefined;
+};
+
+const PortalHomePage = ({
+  user,
+  authToken,
+  pastEvents,
+  upcomingEvents,
+  liveEvents,
+  attendances,
+  checkInResponse,
+}: HomePageProps) => {
+  const [points, setPoints] = useState<number>(user.points);
+  const [attendance, setAttendance] = useState<PublicAttendance[]>(attendances);
+
+  const checkin = async (token: string, attendanceCode: string): Promise<void> => {
+    const response = await attendEvent({ token, attendanceCode });
+    const event = processCheckInResponse(response);
+    if (event) {
+      // This client side code prevents us from having to refetch user and attendance from the API.
```

Comment:
Membership and attendance are being updated, I'm not updating the cookie since I think we agreed on just pulling from the API to fetch user info like membership points for a page (if we rely on the cookie then across different devices it won't be accurate)

Reply:
This is mainly just a little piece so info is visually accurate right after checking in to an event without having to re-call the API. If we traveled to another page where we needed the info, I'd assume we'd re-fetch